### DISCUSSION TITLE
templates/types?

### DISCUSSION THREAD

**Discussion by tessus** [dani-garcia/vaultwarden#2780] [Q&A] [answered]
In the config file I found the following section:

```
## Templates data folder, by default uses embedded templates
## Check source code to see the format
# TEMPLATES_FOLDER=/path/to/templates
```

I checked the documentation of Bitwarden, I searched for `template` and `types` in the Bitwarden and Vaultwarden source code, but I couldn't find anything.

Can anyone explain to me what these templates are for and how to create them? Can I use them to create new types? The 4 types (Login, Card, Identity, Secure Note) are often not enough, thus I would like to create and use a few more.

**Comment by jjlin** [comment]
Templates refers to the Handlebars templates under https://github.com/dani-garcia/vaultwarden/tree/main/src/static/templates. This is mainly useful for customizing email notifications.

Additional item types require both client and server support. Upstream is supposedly working on adding this functionality; see https://community.bitwarden.com/t/bitwarden-roadmap/12865.

**Reply by tessus** [reply]
Thanks a bunch for the info.

I hoped to be able to add more types. I suppose I will have to wait... ;-)

**Comment by stefan0xC** [comment, marked as answer]
I think this is a Vaultwarden specific feature and with this you can customize the templates of the admin panel and the emails.

How? By putting the corresponding handlebar template files you wish to overwrite in the same location in the TEMPLATES_FOLDER, relative to [`src/static/templates`](https://github.com/dani-garcia/vaultwarden/tree/main/src/static/templates). For example, if you want to overwrite [`src/static/templates/email/smtp_test.html.hbs`](https://github.com/dani-garcia/vaultwarden/blob/main/src/static/templates/email/smtp_test.html.hbs) you create `templates/email/smtp_test.html.hbs` in your DATA_FOLDER (which defaults to `data/`) and restart vaultwarden.

**Reply by tessus** [reply]
Thanks. This is also very useful info.
